### DISCUSSION TITLE
No test is going through

### DISCUSSION THREAD

**Discussion by 0xclarus** [Cyfrin/foundry-full-course-cu#1709] [General]
Someone, please help why is none of my test script going through, they keep saying invalid data as shown below.

Test
```
contract FundMeTest is Test {
    FundMe fundMe;
    uint256 num = 20;

    function setUp() external {
        fundMe = new FundMe();
    }

    function testMinimumDollarIsFive() public {
        console.log("minimum usd ", fundMe.MINIMUM_USD());
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }

    function testNum() public {
        assertEq(num, 20);
    }

    function testOwnerIsMsgSender() public {
        assertEq(fundMe.i_owner(), msg.sender);
    }
}
``` error
```
Encountered 3 failing tests in test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: Invalid data] testMinimumDollarIsFive() (gas: 8320)
[FAIL. Reason: Invalid data] testNum() (gas: 5237)
[FAIL. Reason: Invalid data] testOwnerIsMsgSender() (gas: 10634)
```
i even wrote a test num to check even did console.log to see value, and i found out its correct but all test value keeps showing INVALID DATA errors no matter what the test is

**Comment by EngrPips** [comment]
Hello @0xclarus, Please run `foundryup` then retry.

**Reply by 0xclarus** [reply]
I think this worked to make tests work. 

Ty so much, Appreciate. @EngrPips. But, what does foundryup actually do different

**Reply by EngrPips** [reply]
`foundryup` ensures that your foundry software is updated.

**Comment by Pedrog75** [comment]
Hi @0xclarus, 
I ran your code in my environnement, in the way your code is built it seems your are missing some parts: 
You need an address for your FundMe contract instance. If you go back to your FundMe contract you will see the constructor populated with an address priceFeed from AggregatorV3Interface, the s_priceFeed will be used to check if a user is funding the at least the minimum amount of Eth.

After you are calling the function i_owner() from the fundMe instance, in my side the FundMe contract has a getOwner() function whose return an i_owner (address variable). 

<img width="787" alt="Capture d’écran 2024-05-14 à 08 50 06" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/23355828/ae87e4dc-b176-4c08-b2de-c2faaaa82342">

I hope this helps.

**Reply by 0xclarus** [reply]
Yes, this worked, I forgot to set the address in place before running tests especially the testOwnerIsMsgSender() test . TY @Pedrog75 for the assistance
